src/web/demo.rkt

```diff
@@ -258,51 +258,19 @@
   (semaphore-post sema)]
   [(and (*demo-output*) (directory-exists? (build-path (*demo-output*) path)))
   (semaphore-post sema)]
-  [else (match (work-info job-info)
-    [(list 'improve formula)
-      (wrapper-run-herbie 
-       (run-herbie-command 'improve formula (get-seed) #f #f #f) 
-       job-id after-improve)]
-    [(list 'sample formula seed*)
-      (wrapper-run-herbie 
-       (run-herbie-command 'sample formula seed* #f #f #t) 
-       job-id default-after)]
-    [(list 'errors formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'errors formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'exacts formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'exacts formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'evaluate formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'evaluate formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'local-error formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'local-error formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'alternatives formula seed* pcontext)
-      (wrapper-run-herbie 
-       (run-herbie-command 'alternatives formula seed* pcontext #f #t) 
-       job-id default-after)]
-    [(list 'cost formula)
-      (wrapper-run-herbie 
-       (run-herbie-command 'cost formula #f #f #f #t) 
-       job-id default-after)])])
+  [else (wrapper-run-herbie (work-job job-info) job-id (work-after job-info))])
  (eprintf "Job ~a complete\n" job-id)
  (hash-remove! *jobs* job-id)
  (semaphore-post sema))
 
 ; Handles semaphore and async part of a job
-(struct work (id info sema))
+(struct work (id job sema after))
```

Comment:
Given that only one type of command actually uses `after`, I wonder if it could be done somewhere else, just to make the design nicer. Could be another PR.

Reply:
Ya that's a great point. I'm still not sure how to handle the pre and post processing. I imagine it needs some sort of hook or notify when it's done. In one of the other PR's you mentioned returning the `job` when it's done so maybe I could use that to handle this kind of work.